garrysmod/lua/includes/extensions/vector.lua

```diff
@@ -3,10 +3,176 @@ local meta = FindMetaTable( "Vector" )
-	Converts Vector To Color - alpha precision lost, must reset
+Converts Vector to Color - alpha precision lost, must reset
 -----------------------------------------------------------]]
-function meta:ToColor( )
+function meta:ToColor()
+	return Color(self.x * 255, self.y * 255, self.z * 255)
+end
+
+--[[---------------------------------------------------------
+Converts Vector to Table - removes all metrods from the copy
+-----------------------------------------------------------]]
+function meta:ToTable(arr)
+	if(arr) then return {self.x, self.y, self.z} end
+	return {x = self.x, y = self.y, z = self.z}
+end
+
+--[[---------------------------------------------------------
+Converts Vector to Angle - data is copied
+-----------------------------------------------------------]]
+function meta:ToAngle()
+	return Angle(self.x, self.y, self.z)
+end
+
+--[[---------------------------------------------------------
+Returns a copy of a rotated vector. Kind of like v:GetNormalized()
+-----------------------------------------------------------]]
+function meta:GetRotated(ang)
+	local v = Vector(self)
+	v:Rotate(ang)
+	return v
+end
+
+--[[---------------------------------------------------------
+Returns a copy of an added vector
+-----------------------------------------------------------]]
+function meta:GetAdd(vec)
+	local v = Vector(self)
+	v:Add(vec)
+	return v
+end
+
+--[[---------------------------------------------------------
+Returns a copy of a subracted vector
+-----------------------------------------------------------]]
+function meta:GetSub(vec)
+	local v = Vector(self)
+	v:Sub(vec)
+	return v
+end
+
+--[[---------------------------------------------------------
+Returns a copy of a multiplied vector
+-----------------------------------------------------------]]
+function meta:GetMul(num)
+	local v = Vector(self)
+	v:Mul(num)
+	return v
+end
+
+--[[---------------------------------------------------------
+Returns a copy of a divided vector
+-----------------------------------------------------------]]
+function meta:GetDiv(num)
+	local v = Vector(self)
+	v:Div(num)
+	return v
+end
+
+--[[---------------------------------------------------------
+Convets the vector to bisector of two vectors
+-----------------------------------------------------------]]
+function meta:Bisect(vec)
+	local ns, nv = self:Length(), vec:Length()
+	self:Mul(nv)
+	self:Add(vec:GetMul(ns))
+	return self
+end
+
+--[[---------------------------------------------------------
+Returns a copy of bisector of two vectors
+-----------------------------------------------------------]]
+function meta:GetBisected(vec)
+	local v = Vector(self)
+	return v:Bisect(vec)
+end
 
-	return Color( self.x * 255, self.y * 255, self.z * 255 )
+--[[---------------------------------------------------------
+Convets the vector to nagated version of itself
+-----------------------------------------------------------]]
+function meta:Negate()
+	self.x = -self.x

```

Comment:
Unpack here.

Reply:
Done. Added single component trigger. Id only one is needed to be negated. When no arguments, all are negated.